src/main/scala/com/dwolla/lambda/cloudflare/requests/processors/RateLimitProcessor.scala

```diff
@@ -0,0 +1,104 @@
+                       properties: JsonObject): Stream[F, HandlerResponse] =
+    for {
+      request <- parseRecordFrom[WrappedRateLimit](properties, "RateLimit")
+      resp <- handleAction(action, request, physicalResourceId, properties)
+    } yield resp
+
+  private def handleAction(action: CloudFormationRequestType,
+                           request: RateLimit,
+                           physicalResourceId: Option[PhysicalResourceId],
+                           properties: JsonObject,
+                          ): Stream[F, HandlerResponse] = (action, physicalResourceId) match {
+    case (CreateRequest, None) => handleCreate(request, properties)
+    case (UpdateRequest, PhysicalResourceId(zid, prid)) => handleUpdate(zid, prid, request)
+    case (DeleteRequest, PhysicalResourceId(zid, prid)) => handleDelete(zid, prid)
+    case (CreateRequest, Some(id)) => Stream.raiseError(UnexpectedPhysicalId(id))
+    case (_, Some(id)) => Stream.raiseError(InvalidCloudflareUri(id))
+    case (UpdateRequest, None) | (DeleteRequest, None) => Stream.raiseError(MissingPhysicalId(action))
+    case (OtherRequestType(_), _) ⇒ Stream.raiseError(UnsupportedRequestType(action))
+  }
+
+  private def handleCreate(request: RateLimit, properties: JsonObject): Stream[F, HandlerResponse] =
+    for {
+      zone <- parseRecordFrom[String](properties, "Zone")
+      zoneId <- zoneClient.getZoneId(zone)
+      resp <- rateLimitClient.create(zoneId, request)
+    } yield HandlerResponse(tagPhysicalResourceId(resp.id.fold("Unknown RateLimit ID")(id => rateLimitClient.buildUri(zoneId, id))), JsonObject(
+      "created" -> resp.asJson
+    ))
+
+  private def handleUpdate(zoneId: ZoneId, rateLimitId: RateLimitId, request: RateLimit): Stream[F, HandlerResponse] =
+    for {
+      resp <- rateLimitClient.update(zoneId, request.copy(id = Option(rateLimitId)))
+    } yield HandlerResponse(tagPhysicalResourceId(rateLimitClient.buildUri(zoneId, resp.id.getOrElse(rateLimitId))), JsonObject(
+      "updated" -> resp.asJson
+    ))
+
+  private def handleDelete(zoneId: ZoneId, rateLimitId: RateLimitId): Stream[F, HandlerResponse] =
+    for {
+      resp <- rateLimitClient.delete(zoneId, rateLimitId)
+    } yield HandlerResponse(tagPhysicalResourceId(rateLimitClient.buildUri(zoneId, resp)), JsonObject(
+      "deleted" -> resp.asJson
+    ))
+
+  private object PhysicalResourceId {
+    def unapply(arg: Option[PhysicalResourceId]): Option[(ZoneId, RateLimitId)] =
+      arg.flatMap(rateLimitClient.parseUri)
+  }
+
+  object RateLimitProcessor {
+
+    case class WrappedRateLimit(rateLimit: RateLimit)
+
+    object WrappedRateLimit {
+
+      import com.dwolla.cloudflare.domain.model.ratelimits.RateLimit
+
+      // Troposphere by default outputs booleans as strings in its resulting JSON output.
+      // Because of this, we need to update the RateLimit circe decoder so any boolean fields
+      // can be decoded from strings in addition to normal boolean values.
```

Comment:
Troposphere recently fixed this by allowing users to set the `TROPO_REAL_BOOL` environment variable to `true`: https://github.com/cloudtools/troposphere/pull/1409/commits. We could utilize this instead, but I'm not sure if it would have unexpected results for existing users of troposphere (for example `jetstream`).

Reply:
There might also be a better way than I have here to update the `RateLimit` decoder. But, creating a new `WrappedRateLimit` type was the most straightforward way I was able to get changes picked up. For example, alternatives like just adding a new implicitly defined decoder for `RateLimit` did not appear to work.